### ISSUE TITLE
Improve directory structure

### ISSUE THREAD

**Issue by abukaj** [abukaj/kESI#11]
Organization of `extras/FEM/meshes` is less than optimal, while `extras/FEM/solutions` is just a mess.

New structure:
```
data/  # a single mountpoint
  meshes/
    <geometry>/
      <version>/
        <granularity><suffix>
  setups/
    <setup>/
      electrodes.csv
  fenics_leadfield_corrections/
    <setup>/
      <model>==<conductivity>,<geometry>/
        conductivity.csv
        <mesh path>==<version>,<granularity>/
          <degree>/
            <electrode>.h5
            <electrode>.ini
  sampled_leadfield_corrections/
    <setup>/
      <sampling>==<sampling>/ | cropped___<sampling>___<csd basis function>/  # deeper recurrence allowed but not recommended
        grid.npz
        <model>/
          <mesh path>/
            <degree>/
              <electrode>.npz
  potential_basis_functions/
    <setup>/
      <csd basis functions>==<centroids>,<csd profile>
        centroids.npz
        model_src.json
        <path>==kCSD/<conductivity [S/m]>/ |  kESI/<sampling>/<model>/<mesh path>/<degree>/
          <electrode>.npz
  kernels/
    <setup>/
      <subsetup>/
        electrodes.csv
        <csd basis functions>/
          <path>/
            phi.npz
            kernel.npz
            analysis.npz
            <csd grid>/
              grid_csd.npz
              crosskernel.npz

  csd_profiles/
    <setup>/
      <subsetup>/
        <csd basis functions>/
          <path>==kCSD/<conductivity [S/m]>/ |  kESI/<sampling>/<model>/<mesh path>/<degree>/ | mixed/<conductivity [S/m]>/<sampling>/<model>/<mesh path>/<degree>/ 
            <csd grid>/
              <sources>.npz==eigensources.npz
              <fwd path>==<fwd model>/<fwd mesh path>/<fwd degree>/
                <sources>.csv
```

**Comment by abukaj** [comment]
Depends on #14

**Comment by m-kowalska** [comment]
`data` -> `data/generated`

**Comment by abukaj** [comment]
Fixed by dfa914aae11a6505d2b855b877e59523a8507669 - it is almost perfect now. :]
